compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java

```diff
@@ -1974,6 +1980,18 @@ public void resolveConstant(SymbolEnv symEnv, Map<String, BLangNode> modTable, B
         checkUniqueness(constant);
     }
 
+    public void resolveXMLNS(SymbolEnv symEnv, Map<String, BLangNode> modTable, BLangXMLNS xmlnsNode) {
+        if (xmlnsNode.namespaceURI.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+            BLangSimpleVarRef varRef = (BLangSimpleVarRef) xmlnsNode.namespaceURI;
+            BLangNode node = modTable.get(varRef.variableName.value);
+            if (node != null && node.getKind() == NodeKind.CONSTANT &&
+                    !resolvedConstants.contains((BLangConstant) node)) {
```

Comment:
IMO, this isn't correct. The reason is, we need to check in the main space as well. Consider the scenario with `xmlns Foo:foo as ex;`, where we didn't add the `modTable` list. In such cases, we need to check in the main space as well.